core/platform/auth/gae_auth_services.py

```diff
@@ -210,8 +232,15 @@ def associate_multi_auth_ids_to_user_ids(auth_id_user_id_pairs):
 
     assoc_by_user_id_models = [
         auth_models.UserAuthDetailsModel(id=user_id, gae_id=auth_id)
-        for auth_id, user_id in python_utils.ZIP(auth_ids, user_ids)
+        for auth_id, user_id, assoc_by_user_id_model in python_utils.ZIP(
+            auth_ids, user_ids,
+            auth_models.UserAuthDetailsModel.get_multi(user_ids))
+        if (assoc_by_user_id_model is None or
```

Comment:
Same question here as for firebase.

Reply:
Ack'd, let's discuss in the other comment thread: https://github.com/oppia/oppia/pull/11588#discussion_r554284761.

Reply:
Resolved offline, comments + extra checks have been added to explain why this is necessary.